actk/steps/make_diagnostic_sheet/make_diagnostic_sheet.py

```diff
@@ -0,0 +1,294 @@
+
+        # Index to help loop through figure numbers
+        index = 0
+
+        # Set font size
+        font = {"weight": "bold", "size": 11}
+        matplotlib.rc("font", **font)
+
+        # Loop through figure numbers
+        for i in range(len(figure_numbers)):
+
+            # Get rows and columns of figure
+            columns = int(np.sqrt(subplot_numbers[i]) + 0.5)
+            if np.sqrt(subplot_numbers[i]) != int(np.sqrt(subplot_numbers[i])):
+                rows = columns + 1
+            else:
+                rows = columns
+
+            # Set figure size
+            fig_size_width = columns * 7
+            fig_size_height = rows * 5
+
+            # Set subplots
+            fig, ax_array = plt.subplots(
+                rows, columns, squeeze=False, figsize=(fig_size_height, fig_size_width)
+            )
+
+            # Set title
+            fig.suptitle(f"{metadata}:{figure_numbers[i]}")
+
+            # Second index to check subplots
+            index2 = 0
+
+            for k, ax_row in enumerate(ax_array):
+                for j, axes in enumerate(ax_row):
+                    if index2 < subplot_numbers[i]:
+                        # Load feature to plot if feature
+                        if feature:
+                            with open(dataset["CellFeaturesPath"][index]) as f:
+                                this_cell_features = json.load(f)
+                            title = "CellId: {0}, {1} {2}: {3}".format(
+                                dataset["CellId"][index],
+                                "\n",
+                                feature,
+                                this_cell_features[feature],
+                            )
+                            axes.set_title(title)
+                        else:
+                            axes.set_title(f"CellID: {dataset['CellId'][index]}")
+                        axes.axis("off")
+                        # Read AllProjections Image
+                        img = mpimg.imread(
+                            dataset["CellImage2DAllProjectionsPath"][index]
+                        )
+                        axes.imshow(img)
+                        axes.set_aspect(1)
+                        index += 1
+                        index2 += 1
+                    else:
+                        axes.axis("off")
+
+            log.info(
+                f"Completed diagnostic sheet for : {metadata} {figure_numbers[i]} "
+            )
+
+            # Savefig
+            fig.savefig(figure_paths[i])
+
+    @staticmethod
+    def _collect_group(
+        row_index: int,
+        row: pd.Series,
+        diagnostic_sheet_dir: Path,
+        overwrite: bool,
+        metadata: str,
+    ) -> Union[DiagnosticSheetResult, DiagnosticSheetError]:
+        # Don't use dask for image reading
+        aicsimageio.use_dask(False)
+
+        try:
+            # Get the ultimate end save paths for grouped plot
+            if row["CellImage2DAllProjectionsPath"]:
```

Comment:
Would prefer you using `DatasetFields.CellImage2DAllProjectionsPath` over string. Auto-complete in IDE results in less typos (even though you spelled it correctly here). It also means that if we ever change that column name in the future there is a single source to change it rather than all the various strings.